db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2240 @@
+      // create the source
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_source
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalSource.key,
+              externalSource.source_type_name,
+              externalSource.derivation_group_name,
+              externalSource.valid_at,
+              externalSource.start_time,
+              externalSource.end_time,
+              externalSource.created_at
+          )
+      );
+      System.out.println("FINISHED " + externalSource);
+    }
+  }
+
+  protected void insertExternalEvent(ExternalEvent externalEvent) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_event
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalEvent.key,
+              externalEvent.event_type_name,
+              externalEvent.source_key,
+              externalEvent.derivation_group_name,
+              externalEvent.start_time,
+              externalEvent.duration
+          )
+      );
+    }
+  }
+
+  protected void associateDerivationGroupWithPlan(int planId, String derivationGroupName) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event type
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.plan_derivation_group
+          VALUES ('%s', '%s');
+          """.formatted(planId, derivationGroupName)
+      );
+    }
+  }
+
+  /**
+   * Get all derived events.
+   */
+  protected List<DerivedEvent> getDerivedEvents() throws SQLException {
+    List<DerivedEvent> results = new ArrayList<>();
+    try(final var statement = connection.createStatement()) {
+      var res = statement.executeQuery(
+          // language=sql
+          """
+          SELECT * FROM merlin.derived_events;
+          """
+      );
+      while (res.next()) {
+        results.add(new DerivedEvent(
+            res.getString("event_key"),
+            res.getString("event_type_name"),
+            res.getString("source_key"),
+            res.getString("derivation_group_name"),
+            res.getString("start_time"),
+            res.getString("DURATION"),
+            res.getString("source_range"),
+            res.getString("valid_at")
+        ));
+      }
+    }
+    return results;
+  }
+
+  protected List<DerivedEvent> getDerivedEvents(String dg_name) throws SQLException {
+    List<DerivedEvent> results = new ArrayList<>();
+    try(final var statement = connection.createStatement()) {
+      var res = statement.executeQuery(
+          // language=sql
+          """
+          SELECT * FROM merlin.derived_events %s;
```

Comment:
This code should look like:
```suggestion
          SELECT * FROM merlin.derived_events 
          WHERE derivation_group_name = '%s';
```

And its usage should be updated accordingly.